src/into_dart.rs

```diff
@@ -286,6 +286,45 @@ dart_typed_data_type_trait_impl!(
     DartTypedDataType::Float64 => f64 + free_zero_copy_buffer_f64
 );
 
+macro_rules! isize_usize {
+    ($rust_type:ident, $delegate_target_type:ident) => {
+        impl<const N: usize> IntoDart for [$rust_type; N] {
+            fn into_dart(self) -> DartCObject {
+                let vec: Vec<_> = self.into();
+                vec.into_dart()
+            }
+        }
+
+        impl IntoDart for Vec<$rust_type> {
+            fn into_dart(self) -> DartCObject {
+                let vec: Vec<$delegate_target_type> =
+                    self.into_iter().map(|x| x as _).collect();
+                vec.into_dart()
+            }
+        }
+
+        impl<const N: usize> IntoDart for ZeroCopyBuffer<[$rust_type; N]> {
+            fn into_dart(self) -> DartCObject {
+                let vec: Vec<$rust_type> = self.0.into();
+                ZeroCopyBuffer(vec).into_dart()
+            }
+        }
+
+        impl IntoDart for ZeroCopyBuffer<Vec<$rust_type>> {
+            fn into_dart(self) -> DartCObject {
+                let vec: Vec<$delegate_target_type> =
+                    self.0.into_iter().map(|x| x as _).collect();
+                ZeroCopyBuffer(vec).into_dart()
+            }
+        }
+
+        impl IntoDartExceptPrimitive for Vec<$rust_type> {}
+    };
+}
+
+isize_usize!(isize, i64);
+isize_usize!(usize, u64);
```

Comment:
This is okay, except that it only applies to `Amd64` architecture, so I would add feature gates by `target_pointer_size` for example. and on Amd use u32 (instead of u64)

I believe there's a reference somewhere in the code for usize and isize support with pointer_size gates.

Reply:
I also thought about this before, but then realize that: If we choose i32/u32 for 32bit architecture, and i64/u64 for 64bit, then the user will observe that, they get `Int32LIst` Dart type on 32bit and `Int64List` on 64bit. IMHO this may be suboptimal, since Dart does not have convenient conditional compilation, and users in Dart usually expect a single type.

Reply:
Interesting. Okay let's go with 64 bit for now, if we found any issues in the future, we could add support for 32 bit too.